### PR TITLE
[For 3.0.0] Feature/2930  add username at search user

### PR DESCRIPTION
**Pull request by gbonaspetti**
#2930 

## Checkpoints

The goal of this section is to help code reviewer and merge request author to do required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewer**

- [x] Code is clear enough
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

**For developper**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] Original issue have been updated with a short resume of implemented solution

**For quality team**

- [x] Tested by quality team

### CODE REVIEW

File: frontend/src/container/Dashboard.jsx [hunk 1]
```diff
@@ -308,8 +308,8 @@ class Dashboard extends React.Component {
   handleClickValidateNewMember = async () => {
     const { props, state } = this
 
-    if (state.newMember.nameOrEmail === '') {
-      props.dispatch(newFlashMessage(props.t('Please set a name or email'), 'warning'))
+    if (state.newMember.personalData === '') {
+      props.dispatch(newFlashMessage(props.t('Please set a name, an email or an username'), 'warning'))
```
Comment (raphj) [inline frontend/src/container/Dashboard.jsx:312]:
```suggestion
      props.dispatch(newFlashMessage(props.t('Please set a name, an email or a username'), 'warning'))
```

File: frontend_app_workspace_advanced/i18next.scanner/en/translation.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   "Error while saving new description": "Error while saving new description",
   "Error while saving new role for member": "Error while saving new role for member",
   "Error while fetching known members list": "Error while fetching known members list",
-  "Please set a name or email": "Please set a name or email",
+  "Please set a name, an email or an username": "Please set a name, an email or an username",
```
Comment (raphj) [inline frontend_app_workspace_advanced/i18next.scanner/en/translation.json:11]:
```suggestion
  "Please set a name, an email or a username": "Please set a name, an email or an username",
```

File: frontend_app_workspace_advanced/i18next.scanner/fr/translation.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
   "Error while saving new description": "Erreur lors de la sauvegarde de la description",
   "Error while saving new role for member": "Erreur lors de la sauvegarde du rôle",
   "Error while fetching known members list": "Erreur lors du chargement de la liste des membres connus",
-  "Please set a name or email": "Veuillez saisir le nom ou l'email",
+  "Please set a name, an email or an username": "Veuillez saisir le nom, l'email ou l'identifiant",
```
Comment (raphj) [inline frontend_app_workspace_advanced/i18next.scanner/fr/translation.json:11]:
```suggestion
  "Please set a name, an email or a username": "Veuillez saisir le nom, l'email ou l'identifiant",
```

File: frontend_app_workspace_advanced/i18next.scanner/pt/translation.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
   "Error while saving new description": "Erro ao salvar a nova descrição",
   "Error while saving new role for member": "Erro ao salvar a nova função para o membro",
   "Error while fetching known members list": "Erro ao buscar a lista de membros conhecidos",
-  "Please set a name or email": "Defina um nome ou e-mail",
+  "Please set a name, an email or an username": "Defina um nome, um e-mail ou um identificador",
```
Comment (raphj) [inline frontend_app_workspace_advanced/i18next.scanner/pt/translation.json:11]:
```suggestion
  "Please set a name, an email or a username": "Defina um nome, um e-mail ou um identificador",
```

**Comment by raphj** [comment]
This PR does not really touch the `isEmail` field, but I wondered, what does it mean?

**Comment by gbonaspetti** [comment]
> So renaming nameOrEmail is a good idea but `personalData` seems too broad / vague. I suggest using `login` like they do in the backend (so we get consistency), what do you think?

The idea was to maintain consistency with the same change made in issue #2943 and the term PersonalData comes because the users data is modified via a component with the same name.

**Comment by gbonaspetti** [comment]
> This PR does not really touch the `isEmail` field, but I wondered, what does it mean?

I didn't see the need to change it, now that we have the backend development I will do some tests and see if it works well.

**Comment by raphj** [comment]
To me, personal data could also encompass things like birth date, or any other personal data, and could be entered in this personal data UI component if it were needed one day.

for isEmail I was just asking, not implying that changes are needed :-)

**Comment by raphj** [comment]
Ok, I'm good with personnalData :-)

**Comment by gbonaspetti** [comment]
> for isEmail I was just asking, not implying that changes are needed :-)

I wondered the same thing when I was doing the development, or if I should create an "isUsername" variable, but apparently not.
I made the requested changes in the review, but found a problem while doing the tests (if the username is `theuser` and the user search for `@theuser`, it doesn't work), I'll put the issue back to ToDo to solve this problem and do the automated tests.

**Comment by raphj** [comment]
If isEmail checks whether the input value is an email or a username (and not the display name), maybe we could call it isLogin?

**Comment by gbonaspetti** [comment]
> If isEmail checks whether the input value is an email or a username (and not the display name), maybe we could call it isLogin?

From what I understand, the variable is used to send an invitational email to the user, in case he is not known., so in this case it's only valid for emails.

**Comment by raphj** [comment]
Ok so let's keep it as is :-)

File: frontend_lib/src/helper.js [hunk 5]
```diff
@@ -288,3 +288,11 @@ export const IMG_LOAD_STATE = {
   LOADING: 'loading',
   ERROR: 'error'
 }
+
+export const removeAtInUsername = (username) => {
+  let trimmedUsername = username.trim()
+  if (trimmedUsername.length > 0 && trimmedUsername.includes('@') && trimmedUsername[0] === '@') {
```
Comment (raphj) [inline frontend_lib/src/helper.js:294]:
```suggestion
  if (trimmedUsername.length > 0 && trimmedUsername[0] === '@') {
```

**Comment by metzgegu** [comment]
Tested and verified on branch.

**Pull request merged**